web/event_handler.go

```diff
@@ -234,6 +191,61 @@ func (fh *EventHandler) handleStatusEvent(w http.ResponseWriter, body []byte) {
 	}
 }
 
+func (fh *EventHandler) handleDeploymentInfo(w http.ResponseWriter, body []byte) {
```

Comment:
Custom service name should be also handled here. Consider following situation. 
1. Deployed app `A` with label `consul=true`
2. Changed app label to `consul=B`

You end up with two services in consul: A and B.